pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -149,6 +221,42 @@ func (c *clusterMemberRemovalController) sync(ctx context.Context, _ factory.Syn
 	return nil
 }
 
+func (c *clusterMemberRemovalController) removeMemberPendingDeletion(ctx context.Context, memberMachines []*machinev1beta1.Machine, memberType string) error {
+	memberMachinesPendingDeletion := ceohelpers.FilterMachinesPendingDeletion(memberMachines)
+	if len(memberMachinesPendingDeletion) == 0 {
+		return nil
+	}
+
+	// sort by deletion time and pick up one machine for removal and record it
+	sort.Slice(memberMachinesPendingDeletion, func(i, j int) bool {
+		return memberMachinesPendingDeletion[i].DeletionTimestamp.Before(memberMachinesPendingDeletion[j].DeletionTimestamp)
+	})
+	memberMachineToDelete := memberMachinesPendingDeletion[0]
```

Comment:
I expected to iterate over the list of member machines, returning after removing a single member.  otherwise, a single stuck member will prevent any progress.

Reply:
In that scenario, we might remove multiple members, possibly losing quorum. Stable input prevents that. The number of current voting is calculated from the cache. Does it make sense?